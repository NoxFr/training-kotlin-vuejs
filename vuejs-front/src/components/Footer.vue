<template>
  <v-footer height="auto">
      <v-layout row wrap justify-center class="grey darken-3">
          <v-btn
            v-for="icon in icons"
            :key="icon.id"
            icon
            v-bind:href="icon.link"
          >
            <v-icon size="24px">{{ icon.class }}</v-icon>
          </v-btn>
        <v-flex xs12 py-3 text-xs-center white--text>
            &copy;2018 â€” <strong>Beers</strong>
        </v-flex>
      </v-layout>
  </v-footer>
</template>

<script>
export default {
  name: 'tkvFooter',
  data: () => ({
    icons: [ {
      id: 1,
      class: 'fab fa-github',
      link: 'https://github.com/NoxFr/training-kotlin-vuejs'
    },
    {
      id: 2,
      class: 'fab fa-twitter',
      link: 'http://twitter.com/NoxFr'
    },
    {
      id: 3,
      class: 'fab fa-linkedin',
      link: 'https://www.linkedin.com/in/mathieu-durand-260104137/'
    }]
  })
}
</script>
