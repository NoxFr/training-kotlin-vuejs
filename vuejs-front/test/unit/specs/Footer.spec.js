import { mount, createLocalVue } from '@vue/test-utils'
import Footer from '@/components/Footer'
import Vuetify from 'vuetify';

describe('Footer.vue', () => {
  let wrapper
  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(Vuetify); 
  
    wrapper = mount(Footer, { 
      localVue: localVue
    });
  });

  it('should match snapshot', () => {
    // then
    expect(wrapper.element).toBeDefined();
    expect(wrapper.element).toMatchSnapshot();
  })
})
