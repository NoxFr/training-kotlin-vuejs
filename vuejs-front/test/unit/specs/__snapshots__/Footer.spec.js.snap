// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue should match snapshot 1`] = `
<footer
  class="footer"
  style="height: auto;"
>
  <div
    class="layout grey darken-3 row wrap justify-center"
  >
    <a
      class="btn btn--icon"
      href="https://github.com/NoxFr/training-kotlin-vuejs"
    >
      <div
        class="btn__content"
      >
        <i
          aria-hidden="true"
          class="icon fab fa-github"
          style="font-size: 24px;"
        />
      </div>
    </a>
    <a
      class="btn btn--icon"
      href="http://twitter.com/NoxFr"
    >
      <div
        class="btn__content"
      >
        <i
          aria-hidden="true"
          class="icon fab fa-twitter"
          style="font-size: 24px;"
        />
      </div>
    </a>
    <a
      class="btn btn--icon"
      href="https://www.linkedin.com/in/mathieu-durand-260104137/"
    >
      <div
        class="btn__content"
      >
        <i
          aria-hidden="true"
          class="icon fab fa-linkedin"
          style="font-size: 24px;"
        />
      </div>
    </a>
     
    <div
      class="flex xs12 py-3 text-xs-center white--text"
    >
      
          ©2018 — 
      <strong>
        Beers
      </strong>
    </div>
  </div>
</footer>
`;
